marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -114,7 +114,7 @@ public void awaitCompletion() {
                                 getEndpointPath(), getCallCount(), getWorkUnit());
                         break calling;
                     case RUNNING:
-                        if (output == null) {
+                        if (output == null || output.length == 0) {
```

Comment:
The defensive programming here seems correct.

I'm wondering whether arrayCall() should be returning an empty array or null.  Something to discuss.